src/bzfs/bzfsAPI.cxx

```diff
@@ -3272,13 +3267,24 @@ BZF_API void bz_CustomZoneObject::handleDefaultOptions(bz_CustomMapObjectInfo *d
     {
         if (box)
         {
-            xMin = _pos[0] - _size[0];
-            xMax = _pos[0] + _size[0];
-            yMin = _pos[1] - _size[1];
-            yMax = _pos[1] + _size[1];
             zMin = _pos[2];
```

Comment:
I wonder if it is not better for the whole test to translate the position to test in a coordinate sistem where 0,0 is the middle of the box or cylinder, and then, in case of the box to anti-rotate the point against the 0,0 coordinate.

The test then is very easy to do